lib/gpio/src/LimitSwitch.cpp

```diff
@@ -0,0 +1,20 @@
+#include "LimitSwitch.h"
+
+using namespace GPIO;
+
+LimitSwitch::LimitSwitch(DigitalIn limitPin, bool ActiveHigh) : m_limitPin(limitPin), m_ActiveHigh(ActiveHigh) {}
+
+bool LimitSwitch::isPressed() {
+  // Make sure Pin is connected before checking if it's an active high or active low
+  if (bool(m_limitPin)) {
+    if (!m_ActiveHigh) {
+      return !m_limitPin.read();
+    }
+    return m_limitPin.read();
+  }
+  return false;
+}
+
+LimitSwitch::operator bool() {
+  return m_limitPin.is_connected();
```

Comment:
imo the operator bool overload should return whether the limit switch is activated instead of whether it's connected

Reply:
Thank you, I'll try and figure that out